subsys/storage/flash_map/flash_map_default.c

```diff
@@ -7,15 +7,17 @@
 
 #include <zephyr.h>
 #include <storage/flash_map.h>
+#include <flash_pm.h>
+
+#define FLASH_AREA_GEN(lbl)				\
+	{.fa_id = FLASH_AREA_ID(lbl),			\
+	 .fa_off = FLASH_AREA_OFFSET(lbl),		\
+	 .fa_dev_name = FLASH_AREA_DEV_LABEL(lbl),	\
+	 .fa_size = FLASH_AREA_SIZE(lbl),},
 
-#define FLASH_AREA_FOO(i, _)				\
-	{.fa_id = i,					\
-	 .fa_off = DT_FLASH_AREA_##i##_OFFSET,		\
-	 .fa_dev_name = DT_FLASH_AREA_##i##_DEV,	\
-	 .fa_size = DT_FLASH_AREA_##i##_SIZE,},
 
 const struct flash_area default_flash_map[] = {
-	UTIL_LISTIFY(DT_FLASH_AREA_NUM, FLASH_AREA_FOO, ~)
+	UTIL_FOR_LIST(FLASH_AREA_GEN, FLASH_AREA_LIST)
```

Comment:
I checked the preprocessor output with MACRO_MAP instead and it's the same, so I think UTIL_FOR_LIST is not needed

Reply:
Sorry, somehow I missed the `MACRO_MAP` thing. Btw. is counterintuitive that it takes all arguments under `...` instead of having fixed argument for function, followed by `...`.

Reply:
Moved to `MACRO_MAP` macros.